app/main/NameTag.tsx

```diff
@@ -101,11 +101,14 @@ function NameTag({
       <div>
         <h2 style={{ fontSize: '30px', fontWeight: 'bold', display: 'inline-block' }}>Name Tag</h2>
         <ColoredSwitch
-          role='switch'
-          data-testid = 'colored-switch'
-          aria-checked={showNametag}
-          checked={showNametag}
-          onChange={() => setShowNametag(!showNametag)}
+          inputProps={{
+            role: 'switch',
+            'aria-checked': showNametag,
+            checked: showNametag,
+            value: showNametag.toString(),
+            onChange: () => setShowNametag(!showNametag)
+          }}
+          checked={showNametag} // It ensures the switch visually changes when clicked
```

Comment:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

It looks like the checked state is set 4 different times in the code.  I think it should only be set at most once (doesn't a checkbox already keep track of it's own state)?